bob/src/api/mod.rs

```diff
@@ -313,31 +326,42 @@ async fn status<A: Authenticator>(bob: Extension<BobServer<A>>) -> Json<Node> {
 async fn get_space_info<A: Authenticator>(
     bob: Extension<BobServer<A>>,
 ) -> Result<Json<SpaceInfo>, StatusExt> {
-    let DiskSpaceMetrics {
-        total_space,
-        used_space,
-        free_space,
-    } = bob.grinder().hw_counter().update_space_metrics();
-
+    let disk_metrics = bob.grinder().hw_counter().update_space_metrics();
+    let total_disks_metrics: DiskSpaceMetrics = disk_metrics.values().sum();
     let backend = bob.grinder().backend().inner();
     let (dcs, adc) = backend
         .disk_controllers()
         .ok_or_else(not_acceptable_backend)?;
-    let mut map = HashMap::new();
-    for dc in dcs.iter() {
-        map.insert(dc.disk().name().to_string(), dc.disk_used().await);
-    }
+    
+    let mut occupied_disk_space_by_disk: HashMap<DiskName, u64> = futures::future::join_all(
+        dcs.iter().map(|dc| async { ( dc.disk().name().clone(), dc.disk_used().await ) })
+    ).await.into_iter().collect();
+
+    let disk_path: HashMap<_, _> = dcs.iter().map(|dc| (dc.disk().name().to_string(), dc.disk().path().to_string())).collect();
+
+    let disk_space_by_disk = disk_metrics.into_values().map(|disk| (
+        disk_path.get(&disk.disk_name.to_string()).unwrap_or(&disk.disk_name.to_string()).clone(), 
+        Space {
+            total_disk_space_bytes: disk.total_space,
+            free_disk_space_bytes: disk.free_space,
+            used_disk_space_bytes: disk.used_space 
+        }
+    )).collect();
+
     let adc_space = adc.disk_used().await;
```

Comment:
Move the code that adds entry for aliens above to the place where `occupied_disk_space_by_disk` is initialized (line 339)